quickjs.c

```diff
@@ -1313,7 +1299,31 @@ static JSValue js_uint32(uint32_t v)
 {
     if (v <= INT32_MAX)
         return js_int32(v);
-    return js_float64(v);
+    else
+        return js_float64(v);
+}
+
+static JSValue js_int64(int64_t v)
+{
+    if (v == (int32_t)v)
```

Comment:
Maybe write this with explicit checks instead because signed narrowing of a too large value is implementation-defined behavior.
```suggestion
    if (v >= INT32_MIN && v <= INT32_MAX)
```
edit: maybe fix that in quickjs.h too